README.md

```diff
@@ -1,86 +1,47 @@
-It is possible to change the command-line arguments by modifying the `Opts` struct.
-Before launching the service, the `main` function binds to the specified address and prints out the address that was bound to.
-
-The `src/protocol` file contains the enum `Message`, which can include messages with very different structures. 
-The `Message` must provide the `Serialize` and `Deserialize` traits, which are used by the [serde](https://docs.serde.rs/serde/) crate.
-In the application, the `Message` enum includes an `Echo` message that has a (`String`) payload and timestamp; it also includes
-`Heartbeat` and `HeartbeatAck` messages that carry no information other than their type. (Messages are delivered with the sender 
-address attached, so these empty message types can be useful.) 
-
-The `src/helpers.rs` file contains any helper functions that are needed. In our example this is just a function to print
-a dataflow graph representation of the hydroflow code to stdout.
-
-Each service file comes with a skeleton Hydroflow spec that provides an inbound communication channel and an outbound communication channel, both 
-bound to the specified address and port. The channels are named `inbound_chan` and `outbound_chan`, and are accessed in hydroflow code using the 
-`source_serde` and `sink_serde` operators respectively. The single address/port pair is sufficient in general to support multiple
-different `Message` types across multiple services and instances. Upon receipt, messages are handled by the appropriate code using hydroflow's `demux`
-operator. It is also possible to open more channels to segregate traffic to different IP addresses or ports, simply by copying the 
-patterns that define and use the `inbound` and `outbound` channels.
+## Template Project Structure
+The `src` directory contains the following files:
 
-Each service file also includes code to generate the dataflow graph for the service, if the `--graph` flag is provided on the command line.
-The ASCII spec for the graph is printed to stdout on launch.
+| File          | Description                                                                                                                          | 
+|---------------|--------------------------------------------------------------------------------------------------------------------------------------|
+| `main.rs`     | Contains `main` entry-point function for both client and server. Performs command-line argument parsing.                             |
+| `protocol.rs` | Contains the `Message` enum that defines the messages that can be sent between instances.                                            |
+| `<role>.rs`   | Contains the service for the given role. Example implementations and skeletal hydroflow spec are provided for `server` and `client`. |
+| `helpers.rs`  | Contains helper functions that are invoked from Hydroflow code in multiple services.                                                 |
```

Comment:
I hope y'all are OK with tables. RustRover markdown editor makes tables really easy, but if y'all feel strongly about not dealing with tables, I can revert this back to the original text blcok.

Reply:
Tables are 👌